### DISCUSSION TITLE
How to convert Assembly to Volume?

### DISCUSSION THREAD

**Discussion by jaydeepsb** [marcomusy/vedo#672] [Q&A] [answered]
Hello, I am trying to create different 3D shape - sphere, cube, pyramid etc. And I wish to convert it to Volume (and also to numpy array).
However, my steps below result in an error - 

`import vedo`
`from vedo import Volume`
`a = vedo.Sphere(r=10, res=100) + vedo.Sphere(r=9, res=100) `
`a_vol = vedo.Volume(a)` 
`a_vol.tonumpy()`

AttributeError: 'NoneType' object has no attribute 'GetDimensions'
Any idea where I am gettign wrong?

**Comment by marcomusy** [comment, marked as answer]
Hi, sorry you mean creating a Volume where inside voxels are set to 1 and outer voxels are set to zero? then you can use `.binarize()`
```python
from vedo import Volume, Sphere, merge, show
a = merge(Sphere(r=12, res=100), Sphere(r=9, res=100))
a_vol = a.binarize(spacing=(0.2,0.2,0.2))
arr = a_vol.tonumpy()
print(arr.shape)
show(a_vol)
```
![Screenshot from 2022-07-21 13-12-33](https://user-images.githubusercontent.com/32848391/180200496-817f8546-0dd9-44ee-83c9-7f0b0d6093ae.png)

**Reply by jaydeepsb** [reply]
Thank you! binarize works for my case.
